### PR TITLE
Support message sender(icon/nickname switch)

### LINKED ISSUE #274: Support icon-nickname-switch
News: https://developers.line.biz/en/news/2020/03/17/icon-nickname-switch/
Doc: https://developers.line.biz/en/docs/messaging-api/icon-nickname-switch/

### PR DESCRIPTION
**Pull request by snmatsui**
https://developers.line.biz/en/docs/messaging-api/icon-nickname-switch/
Closes #274

### CODE REVIEW

File: tests/LINEBot/SenderTest.php [hunk 1]
```diff
@@ -0,0 +1,352 @@
+<?php
+/**
+ * Copyright 2018 LINE Corporation
```
Comment (moririnson) [inline tests/LINEBot/SenderTest.php]: Change 2018 → 2020.

Comment (snmatsui) [reply tests/LINEBot/SenderTest.php]: I fixed copyrights in new classes

File: src/LINEBot/SenderBuilder/SenderMessageBuilder.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+use LINE\LINEBot\SenderBuilder\SenderBuilder;
+
+/**
+ * A builder class for sender message.
+ *
+ * @package LINE\LINEBot\SenderMessageBuilder
+ */
+class SenderMessageBuilder implements SenderBuilder
+{
+    /** @var string */
```
[the remark this answers is not in the document]
Comment (snmatsui) [reply src/LINEBot/SenderBuilder/SenderMessageBuilder.php]: fixed phpdoc comments

File: src/LINEBot/SenderBuilder/SenderMessageBuilder.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    public function buildSender()
+    {
+        if (!is_null($this->sender)) {
+            return $this->sender;
+        }
+
+        $sender = [];
+
+        if (!empty($this->name)) {
+            $sender["name"]=$this->name;
```
[the remark this answers is not in the document]
Comment (snmatsui) [reply src/LINEBot/SenderBuilder/SenderMessageBuilder.php]: fixed these code styles

File: src/LINEBot/SenderBuilder/SenderMessageBuilder.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ * @package LINE\LINEBot\SenderMessageBuilder
+ */
+class SenderMessageBuilder implements SenderBuilder
+{
+    /** @var string */
+    private $name;
+    /** @var string */
+    private $iconUrl;
+    /** @var array */
+    private $sender;
```
[the remark this answers is not in the document]
Comment (snmatsui) [reply src/LINEBot/SenderBuilder/SenderMessageBuilder.php]: I referened other builder classes(ex: QuickReplyMessageBuilder, TextMessageBuilder, etc)
They cache their built array.
So I followed them

[the remark this answers is not in the document]
Comment (moririnson) [reply src/LINEBot/SenderBuilder/SenderMessageBuilder.php]: I see. thanks!

File: src/LINEBot/SenderBuilder/SenderMessageBuilder.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public function buildSender()
+    {
+        if (!is_null($this->sender)) {
+            return $this->sender;
+        }
+
+        $sender = [];
+
+        if (!empty($this->name)) {
```
Comment (Vaduz) [inline src/LINEBot/SenderBuilder/SenderMessageBuilder.php]: You should use `isset` inseted of `empty` to enable users `0` or `"0"` as their name. 
```suggestion
        if (isset($this->name)) {
```

File: tests/LINEBot/SenderTest.php [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+
+    public function testStickerMessageWithSender()
+    {
+        $sender = new SenderMessageBuilder(null, "https://example.com/test2");
+
+        $messageTemplate = new StickerMessageBuilder('1', '1', null, $sender);
+
+        $this->assertEquals(
+            [
+            [
```
Comment (Vaduz) [inline tests/LINEBot/SenderTest.php]: Fix indent please. And also other tests below.

File: src/LINEBot/SenderBuilder/SenderMessageBuilder.php [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            $sender['name'] = $this->name;
+        }
+        if (!empty($this->iconUrl)) {
```
Comment (Vaduz) [inline src/LINEBot/SenderBuilder/SenderMessageBuilder.php]:
```suggestion
        if (isset($this->iconUrl)) {
```

File: tests/LINEBot/SenderTest.php (hunk 6 above)
Comment (snmatsui) [reply tests/LINEBot/SenderTest.php]: I formatted these styles

**Comment by moririnson** [comment]
@snmatsui merged this. thank you!!

**Pull request merged**